src/main/java/org/zalando/nakadi/service/subscription/state/StreamingState.java

```diff
@@ -38,6 +38,8 @@
     private long committedEvents;
     private long sentEvents;
     private long batchesSent;
+    // Uncommitted offsets are calculated right on exiting from Streaming state.
+    private Map<Partition.PartitionKey, Long> uncommittedOffsets;
```

Comment:
I do not check it for nulls anywhere in this case.
Because it is stated, that every state will go thru `onExit` and `onEnter` calls. From the other hand, if I will initialize this field with something incorrect (for example empty), than someone who use it will be able to treat it incorrectly. It is better to return value that will lead to exception than.